src/Client/Web/Bit.Client.Web.BlazorUI.Tests/NumericTextField/BitNumericTextFieldTests.cs

```diff
@@ -0,0 +1,929 @@
+        {
+            var component = RenderComponent<BitNumericTextFieldTest>(parameters =>
+            {
+                parameters.Add(p => p.IsArrowEnable, true);
+            });
+
+            var decreaseButton = component.FindAll("button")[1];
+            var onDecrementEventCounterInitValue = component.Instance.OnDecrementEventCounter;
+            for (int i = 0; i < countOfClicks; i++)
+            {
+                decreaseButton.MouseDown();
+            }
+
+            Assert.AreEqual(onDecrementEventCounterInitValue - countOfClicks, component.Instance.OnDecrementEventCounter);
+        }
+
+        [DataTestMethod,
+           DataRow(4),
+           DataRow(12)
+        ]
+        public void BitNumericTextFieldInputOnBlurEventCallbackTest(int countOfBlur)
+        {
+            var component = RenderComponent<BitNumericTextFieldTest>();
+
+            var input = component.Find("input");
+            for (int i = 0; i < countOfBlur; i++)
+            {
+                input.Blur();
+            }
+
+            Assert.AreEqual(countOfBlur, component.Instance.OnBlurEventCounter);
+        }
+
+        [DataTestMethod,
+           DataRow(4),
+           DataRow(12)
+        ]
+        public void BitNumericTextFieldInputOnFocusTest(int countOfFocus)
+        {
+            var component = RenderComponent<BitNumericTextFieldTest>();
+
+            var input = component.Find("input");
+            for (int i = 0; i < countOfFocus; i++)
+            {
+                input.Focus();
+            }
+
+            Assert.AreEqual(countOfFocus, component.Instance.OnFocusEventCounter);
+        }
+
+        [DataTestMethod,
+           DataRow(4),
+           DataRow(12)
+        ]
+        public void BitNumericTextFieldOnChangeTest(double inputValue)
+        {
+            var component = RenderComponent<BitNumericTextFieldTest>();
+
+            var input = component.Find("input");
+            var changeArgs = new ChangeEventArgs();
+            changeArgs.Value = inputValue;
+            input.Change(inputValue);
+            input.Blur();
+
+            Assert.AreEqual(inputValue, component.Instance.OnChangeEventValue);
+        }
+
+        [DataTestMethod,
+           DataRow(null),
+           DataRow("AriaDescription")
+        ]
+        public void BitNumericTextFieldShouldHaveCorrectAriaDecription(string ariaDescription)
+        {
+            var component = RenderComponent<BitNumericTextFieldTest>(parameters =>
+            {
+                parameters.Add(p => p.AriaDescription, ariaDescription);
+            });
+
+            var input = component.Find("input");
+
+            Assert.AreEqual(ariaDescription, input.GetAttribute("aria-describedby"));
+        }
+
+        [DataTestMethod,
+           DataRow(3.0, null),
+           DataRow(3.0, " cm"),
+           DataRow(null, null)
+        ]
+        public void BitNumericTextFieldInputShouldHaveCorrectAriaValueNow(double? ariaValueNow, string suffix)
```

Comment:
In my opinion, it is better to use decimal type in all tests instead of double type.

Reply:
It's not possible because decimal is not a primitive type so if changed from double to decimal all test has decimal as a parameter failed. The reason of the failure is `DataRow` can't cast the input to a none-primitive type.

Reply:
interesting. we need to discucss it ASAP then.
@Cyrus-Sushiant + @milad-karimifard